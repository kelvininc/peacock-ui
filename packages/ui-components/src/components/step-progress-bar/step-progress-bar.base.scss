@import '../../assets/styles/globals';

:host {
	/**
	* @prop --progress-bar-height: The height of the progress bar
	* @prop --progress-bar-background-color: The color of the progress bar container
	* @prop --progress-bar-filler-success-color: The color of the progress bar filler
	* @prop --progress-bar-filler-error-color: The color of the progress bar filler when in error
	*/
	--progress-bar-height: 12px;
}

$progress-bar-height-half: calc(var(--progress-bar-height) / 2);

.progress-wrapper {
	display: flex;
	position: relative;
	align-items: center;
	justify-content: space-between;
	height: var(--progress-bar-height);
	border-radius: $progress-bar-height-half;
	background-color: var(--progress-bar-background-color);
	padding: 0 $progress-bar-height-half;

	.progress-bar-filler {
		position: relative;
		height: inherit;
		background-color: var(--progress-bar-filler-success-color);
		transition: width 0.25s ease-in-out;

		&.error {
			background-color: var(--progress-bar-filler-error-color);
		}
	}

	.steps-container {
		position: absolute;
		left: 0;
		width: 100%;
		height: 100%;
		display: flex;
		justify-content: space-between;
		align-items: center;
	}
}
