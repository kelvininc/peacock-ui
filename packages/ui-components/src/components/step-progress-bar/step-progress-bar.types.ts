export interface IStepProgressBar {
	/** (required) Defines how much space the progress bar should fill */
	progressPercentage: number;
	/** (optional) Defines if the bar should be in an error state, setting to `true` will change the background color to the defined error color */
	hasError: boolean;
}
