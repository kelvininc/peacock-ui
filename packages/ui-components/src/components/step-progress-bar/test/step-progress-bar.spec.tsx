import { SpecPage, newSpecPage } from '@stencil/core/testing';
import { KvStepProgressBar } from '../step-progress-bar';

describe('StepProgressBar (unit tests)', () => {
	let page: SpecPage;
	let component: KvStepProgressBar;

	describe('when rendering with required props', () => {
		beforeEach(async () => {
			page = await newSpecPage({
				components: [KvStepProgressBar],
				html: `
					<kv-step-progress-bar progress-percentage=50>
						<kv-step-indicator active />
						<kv-step-indicator active />
						<kv-step-indicator />
					</kv-step-progress-bar>
				`
			});
			component = page.rootInstance;
		});

		describe('when the component loads', () => {
			it('should match the snapshot', () => {
				expect(page.root).toMatchSnapshot();
			});

			it('should assign the correct value to the `progressPercentage` prop', () => {
				expect(component.progressPercentage).toEqual(50);
			});
		});
	});

	describe('when rendering with required props and error state', () => {
		beforeEach(async () => {
			page = await newSpecPage({
				components: [KvStepProgressBar],
				html: `
					<kv-step-progress-bar progress-percentage=50 has-error>
						<kv-step-indicator active has-error />
						<kv-step-indicator active has-error />
						<kv-step-indicator />
					</kv-step-progress-bar>
				`
			});
			component = page.rootInstance;
		});

		describe('when the component loads', () => {
			it('should match the snapshot', () => {
				expect(page.root).toMatchSnapshot();
			});

			it('should assign the correct value to the `progressPercentage` prop', () => {
				expect(component.progressPercentage).toEqual(50);
			});

			it('should assign the correct value to the `hasError` prop', () => {
				expect(component.hasError).toEqual(true);
			});
		});
	});
});
