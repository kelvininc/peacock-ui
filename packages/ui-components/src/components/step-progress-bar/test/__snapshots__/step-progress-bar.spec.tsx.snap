// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepProgressBar (unit tests) when rendering with required props and error state when the component loads should match the snapshot 1`] = `
<kv-step-progress-bar has-error="" progress-percentage="50">
  <mock:shadow-root>
    <div class="progress-wrapper">
      <div class="error progress-bar-filler" style="width: 50%;"></div>
      <div class="steps-container">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  <kv-step-indicator active="" has-error="">
    <kv-step-indicator active="" has-error="">
      <kv-step-indicator></kv-step-indicator>
    </kv-step-indicator>
  </kv-step-indicator>
</kv-step-progress-bar>
`;

exports[`StepProgressBar (unit tests) when rendering with required props when the component loads should match the snapshot 1`] = `
<kv-step-progress-bar progress-percentage="50">
  <mock:shadow-root>
    <div class="progress-wrapper">
      <div class="progress-bar-filler" style="width: 50%;"></div>
      <div class="steps-container">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  <kv-step-indicator active="">
    <kv-step-indicator active="">
      <kv-step-indicator></kv-step-indicator>
    </kv-step-indicator>
  </kv-step-indicator>
</kv-step-progress-bar>
`;
